src/main/java/pl/touk/flink/ignite/dialect/IgniteDialect.java

```diff
@@ -31,9 +30,72 @@ public String getLimitClause(long l) {
         return "LIMIT " + l;
     }
 
+    @Override
+    public String quoteIdentifier(String identifier) {
+        return "\"" + identifier + "\"";
+    }
+
     @Override
     public Optional<String> defaultDriverName() {
         return Optional.of("org.apache.ignite.IgniteJdbcThinDriver");
     }
 
+    // WARNING code below was taken from PostgresDialect, it can turn out to be wrong
+
+    private static final int MAX_TIMESTAMP_PRECISION = 6;
+    private static final int MIN_TIMESTAMP_PRECISION = 1;
+
+    private static final int MAX_DECIMAL_PRECISION = 1000;
+    private static final int MIN_DECIMAL_PRECISION = 1;
+
+    @Override
+    public Optional<String> getUpsertStatement(
+            String tableName, String[] fieldNames, String[] uniqueKeyFields) {
+        String uniqueColumns =
+                Arrays.stream(uniqueKeyFields)
+                        .map(this::quoteIdentifier)
+                        .collect(Collectors.joining(", "));
+        String updateClause =
+                Arrays.stream(fieldNames)
+                        .map(f -> quoteIdentifier(f) + "=EXCLUDED." + quoteIdentifier(f))
+                        .collect(Collectors.joining(", "));
+        return Optional.of(
+                getInsertIntoStatement(tableName, fieldNames)
+                        + " ON CONFLICT ("
+                        + uniqueColumns
+                        + ")"
+                        + " DO UPDATE SET "
+                        + updateClause);
+    }
+
+    @Override
+    public Optional<Range> decimalPrecisionRange() {
+        return Optional.of(Range.of(MIN_DECIMAL_PRECISION, MAX_DECIMAL_PRECISION));
```

Comment:
Can you extend tests with a new column of decimal type? It is supported by Ignite: https://ignite.apache.org/docs/latest/sql-reference/data-types#decimal

Reply:
Whoops, DECIMAL type is already tested :)